### PR TITLE
WEB-79: Frontend Google Sign-in

### PR DESCRIPTION
**Pull request by NWCalvank**
Adds toggle-able support for Google Sign-in.
- Basic Public/Private routes
- Application-level state that tracks user's authentication state + session token
- Automatically "sign-in" authenticated users

The hope is that this implementation is as simple-to-understand/extend as possible. For example, there is no complexity around redirecting, and redux has not been added. This has drawbacks for a long-term project, but it should be beneficial for time-sensitive POCs.

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -44,6 +44,43 @@ The following sample components are provided:
...
+### Google Sign-in Support
+
+#### Setup
+
+Some quick POCs may require an integration with Google Sign-in. This support is
+toggled off by default, but it can be easily enabled in the `App` component.
+
+Past projets have been set up on [Google Developer
+Console](https://console.developers.google.com/), so you may be able to reach
+out for an existing `client_id`. This value is required to be set in the `.env`
```
Comment (fwkendall) [inline client/README.md:56]: I guess my question is, who do I reach out to?

IIRC Kit created the current SprintWithUs project. Does it need to be Kit now who can provide access? Or is it best to ask in #web-dev and multiple people could act?

**Review by fwkendall — APPROVED** [review]
Looks good! Just one comment in the readme :)

File: client/README.md (hunk 1 above)
Comment (NWCalvank) [reply client/README.md:56]: So, for the SWU that I was on, I set up the `client_id` on my own account. I already knew how to create a project, so it took me <5 minutes. I had heard rumours that other account(s) already existed, which is what I was getting at in this piece of the README.

Do we have a web team Google Developers account? Ideally, we'd just have an ongoing project on a department account, specifically for code challenges/the boilerplate.

If we have such an account, we can likely put the credentials in a safe place and then document where to find them much more explicitly in the README here.

Thoughts?

Comment (fwkendall) [reply client/README.md:56]: I think that ultimately makes the most sense, yeah. That's part of what WEB-55 will hopefully address, so let's just leave it as-is for now until we get to that ticket. :)

**Pull request merged**